api/base/serializers.py

```diff
@@ -328,6 +350,26 @@ def __init__(self, **kwargs):
         self.link_type = kwargs.pop('link_type', 'url')
         super(JSONAPIHyperlinkedGuidRelatedField, self).__init__(read_only=True, **kwargs)
 
+    def resolve(self, resource):
+        """
+        Resolves the view for target node or target comment when embedding.
+        """
+        embed_value = resource.target._id
+        if resource.target._name == 'node':
```

Comment:
Since TargetField is looking for a node or a comment, there's just the one kwarg, either node_id or comment_id, which is set equal to the `resource.target._id`. I don't know if there is a need to make this field more generic, this seems to be a field for a specific case, like CheckoutField.  

Also, it doesn't inherit from RelationshipsField so there is currently no functionality to resolve the angular brackets.